qiskit/circuit/add_control.py

```diff
@@ -126,13 +126,13 @@ def control(operation, num_ctrl_qubits=1, label=None):
     instr = qc.to_instruction()
     if isinstance(operation, controlledgate.ControlledGate):
         new_num_ctrl_qubits = num_ctrl_qubits + operation.num_ctrl_qubits
-        base_name = operation.base_gate_name
+        base_name = operation.base_gate.name
         base_gate = operation.base_gate
-        base_gate_name = operation.base_gate_name
+        base_gate_name = operation.base_gate.name
```

Comment:
we can probably just remove `base_gate_name` now

Reply:
yes